cpp/src/io/orc/orc.cpp

```diff
@@ -411,38 +400,32 @@ const uint8_t* OrcDecompressor::Decompress(const uint8_t* srcBytes, size_t srcLe
       max_dst_length += m_blockSize;
     }
     i += block_len;
-    if (i > srcLen || block_len > m_blockSize) { return nullptr; }
+    CUDF_EXPECTS(i <= src.size() and block_len <= m_blockSize, "Error in decompression");
   }
   // Check if we have a single uncompressed block, or no blocks
-  if (max_dst_length < m_blockSize) {
-    if (srcLen < 3) {
-      // Total size is less than the 3-byte header
-      return nullptr;
-    }
-    *dstLen = srcLen - 3;
-    return srcBytes + 3;
-  }
+  if (max_dst_length < m_blockSize) { return src.subspan(header_size, src.size() - header_size); }
+
   m_buf.resize(max_dst_length);
-  auto dst          = m_buf.data();
   size_t dst_length = 0;
-  for (size_t i = 0; i + 3 < srcLen;) {
-    uint32_t block_len       = srcBytes[i] | (srcBytes[i + 1] << 8) | (srcBytes[i + 2] << 16);
+  for (size_t i = 0; i + header_size < src.size();) {
+    uint32_t block_len       = src[i] | (src[i + 1] << 8) | (src[i + 2] << 16);
     uint32_t is_uncompressed = block_len & 1;
```

Comment:
This seems very much like a `bool`.
```suggestion
    bool const is_uncompressed = static_cast<bool>(block_len & 1);
```